tests/tools/onert_train/src/types.h

```diff
@@ -17,11 +17,58 @@
 #ifndef __ONERT_TRAIN_TYPES_H__
 #define __ONERT_TRAIN_TYPES_H__
 
+#include <type_traits>
+#include <stdexcept>
+
 namespace onert_train
 {
 
 using TensorShape = std::vector<int>;
 
+template <class T, class Enable = void> struct Optional
```

Comment:
IMO, we can use c++17 for `onert_train`. Otherwise, we can use `boost::optional`.

Reply:
Thanks for giving the idea.
For this draft, Let me try to use c++17.